src/tools/rbd/action/Export.cc

```diff
@@ -86,12 +365,156 @@ class C_Export : public Context
   SimpleThrottle &m_throttle;
   librbd::Image &m_image;
   bufferlist m_bufferlist;
+  uint64_t m_dest_offset;
   uint64_t m_offset;
   uint64_t m_length;
   int m_fd;
 };
 
-static int do_export(librbd::Image& image, const char *path, bool no_progress)
+static int do_export_v2(librbd::Image& image, librbd::image_info_t &info, int fd,
+		        uint64_t period, int max_concurrent_ops, utils::ProgressContext &pc)
+{
+  int r = 0;
+  // header
+  bufferlist bl;
+  bl.append(utils::RBD_IMAGE_BANNER_V2);
+
+  // size in header
+  uint64_t size = info.size;
+  ::encode(size, bl);
+
+  __u8 tag;
+  uint64_t length;
+  // encode order
+  tag = RBD_EXPORT_IMAGE_ORDER;
+  length = 8;
+  ::encode(tag, bl);
+  ::encode(length, bl);
+  ::encode(uint64_t(info.order), bl);
+
+  // encode format
+  tag = RBD_EXPORT_IMAGE_FORMAT;
```

Comment:
We are trying to eliminate v1 images (already deprecated) -- so I think this is safe to leave out since it will soon provide zero value.